### PR TITLE
MT76xx_Common: Make the interface names configurable on MediaTek plat…

### PR DESCRIPTION
**Pull request by venkatchimata**
…forms

Fixes #

### Proposed Changes

This pull request:

 -
 -
 -

### CODE REVIEW

File: src/unum/wireless/mt76xx_common/mt76xx_stas_platform.c [hunk 1]
```diff
@@ -106,20 +106,22 @@ int wt_tpl_fill_sta_info(WT_JSON_TPL_RADIO_STATE_t *rinfo,
...
-    } else if(strncmp(vinfo->ifname, "ra", 2) == 0) {
+    } else if(strncmp(vinfo->ifname, WIFI_RADIO_24_PREFIX, 2) == 0) {
         vap_id = vinfo->ifname[2] - '0';
     }
     if(vap_id < 0 || vap_id > 3) {
         log("%s: failed to get VAP id from <%s>\n", __func__, vinfo->ifname);
         return -3;
     }
 
+#if WT_QUERY_STAS_FOR_EACH_VAP
```
Comment (dob71) [inline src/unum/wireless/mt76xx_common/mt76xx_stas_platform.c]: This should be "#ifndef WT_QUERY_STAS_FOR_EACH_VAP"
This define is not a value, it's either "defined" or "not defined".
The ApIdx check is not needed if we query per-VAP (if we have to get STAs per radio then we need that check).

File: src/unum/wireless/mt76xx_common/mt76xx_radios_platform.c [hunk 2]
```diff
@@ -21,11 +21,10 @@
 
 // List of platform radios indexed by the radio number
 static char *radio_list[] = {
-    "ra0", // 2.4 GHz radio
-    "rai0" // 5 GHz radio
+    WIFI_RADIO_24, // 2.4 GHz radio
```
Comment (dob71) [inline src/unum/wireless/mt76xx_common/mt76xx_radios_platform.c]: I don't think we need these defines. You can use:
```
{
WIFI_RADIO_24_PREFIX "0",
WIFI_RADIO_5_PREFIX "0"
}
```

**Review by dob71 — CHANGES_REQUESTED** [review]

**Comment by dob71** [comment]
Also, where are the defaults for those defines?

File: src/unum/wireless/mt76xx_common/mt76xx_radios_platform.c (hunk 2 above)
Comment (venkatchimata) [reply src/unum/wireless/mt76xx_common/mt76xx_radios_platform.c]: Changed.

Comment (venkatchimata) [reply src/unum/wireless/mt76xx_common/mt76xx_radios_platform.c]: There are no default values as of now.

Comment (dob71) [reply src/unum/wireless/mt76xx_common/mt76xx_radios_platform.c]: Ok, but you'll need them (the defaults) here in this pull request. All the external builds using the opensource code should still work without the need to change anything in their repos after you merge this pull request. Maintaining backward compatibility is now required for this code.

File: src/unum/wireless/mt76xx_common/mt76xx_stas_platform.c (hunk 1 above)
Comment (venkatchimata) [reply src/unum/wireless/mt76xx_common/mt76xx_stas_platform.c]: HI Denis, I think it should be #ifdef WT_QUERY_STAS_FOR_EACH_VAP . #ifndef  hits this condition and it fails to retrieve the stations.

File: src/unum/wireless/mt76xx_common/mt76xx_radios_platform.c (hunk 2 above)
Comment (venkatchimata) [reply src/unum/wireless/mt76xx_common/mt76xx_radios_platform.c]: Hi Denis, I think there are no opensource repos based on mt76xx .

Comment (venkatchimata) [reply src/unum/wireless/mt76xx_common/mt76xx_radios_platform.c]: Added default values.

File: src/unum/wireless/mt76xx_common/mt76xx_stas_platform.c (hunk 1 above)
Comment (dob71) [reply src/unum/wireless/mt76xx_common/mt76xx_stas_platform.c]: In this case your platform code must be reporting STAs incorrectly.
You probably need to define WT_QUERY_STAS_FOR_EACH_VAP in your platform code.
That will make mt76xx generic code query STAs from each VAP interface on a radio, for example ra0, ra1 and so on.
Without that define mt76xx code expects the drivers to return STAs for all the VAP interfaces when querying interface 0, i.e. ra0 and after that filters them out by individual VAPs by checking the ApIdx.

File: src/unum/wireless/mt76xx_common/mt76xx_radios_platform.c [hunk 3]
```diff
@@ -3,7 +3,6 @@
 
 #include "unum.h"
 
-
```
Comment (dob71) [inline src/unum/wireless/mt76xx_common/mt76xx_radios_platform.c:6]: Please make sure there are no unnecessary changes.

File: src/unum/wireless/mt76xx_common/mt76xx_stas_platform.c [hunk 4]
```diff
@@ -6,6 +6,7 @@
 // Include declarations for mt76xx
 #include "mt76xx_wireless_platform.h"
 
+
```
Comment (dob71) [inline src/unum/wireless/mt76xx_common/mt76xx_stas_platform.c:9]: Please make sure there are no unnecessary changes.

File: src/unum/wireless/mt76xx_common/mt76xx_wireless_platform.c [hunk 5]
```diff
@@ -4,12 +4,13 @@
 
 #include "unum.h"
 
-// Currently we only need this .h here for klogctl()
```
Comment (dob71) [inline src/unum/wireless/mt76xx_common/mt76xx_wireless_platform.c:7]: Please make sure there are no unnecessary changes.
